src/utils/allocators.zig

```diff
@@ -984,3 +996,137 @@ test "disk allocator large realloc" {
 
     page1[page1.len - 1] = 10;
 }
+
+test "fuzzDiskMemoryAllocator - past failures" {
+    const test_cases = [_]struct { usize, u16, usize }{
+        .{ 18813, 8, 2 },
+        .{ 40309, 8, 2 },
+        .{ 41790, 8, 2 },
+        .{ 47097, 8, 2 },
+        .{ 2952, 8, 10 },
+        .{ 6143, 8, 10 },
+        .{ 10455, 8, 10 },
+        .{ 10887, 8, 10 },
+        .{ 11639, 8, 10 },
+        .{ 79, 8, 100 },
+        .{ 13, 8, 1000 },
+        // .{ 0, 8, 10_000 }, // slow
+    };
+    const debug = false;
+    for (test_cases) |case| {
+        const seed, const mmap_ratio, const iterations = case;
+        var rng = std.Random.DefaultPrng.init(seed);
+        if (debug) std.debug.print("\n>>> Test Case: {}, {}, {}\n", .{ seed, mmap_ratio, iterations });
+        try fuzzDiskMemoryAllocator(.{
+            .allocator = std.testing.allocator,
+            .random = rng.random(),
+            .iterations = iterations,
+            .debug = debug,
+        }, mmap_ratio);
+    }
+}
+
+pub fn runFuzzer(seed: u64, args: *std.process.ArgIterator) !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
```

Comment:
Should we be deiniting this to check for leaks?

Reply:
a699e332aa5c3580bbcdf4f0fb5965d9dd4e7a63

Reply:
16334042818dc88fd3062860202372575093cdda